bytecomp/translattribute.mli

```diff
@@ -58,3 +48,9 @@ val get_and_remove_specialised_attribute
 val get_tailcall_attribute
    : Typedtree.expression
   -> bool * Typedtree.expression
+
+val add_function_attributes
+  : Lambda.lambda
+  -> Location.t
+  -> Parsetree.attributes
+  -> Lambda.lambda
```

Comment:
I'm not sure about this factorization of the `{get,add}_foo_attribute` functions. If we later want to extend some of them to other things than functions (functors, local references, etc.), then we need the per-attribute operations again. I think I would prefer the interface with each operation exposed, and the caller calling the three of them explicitly.

(If you had time to factorize the notion of "operation on attributes", you could define a datatype for that and have a `val function_attributes : attribute_option list` to be called conveniently on the caller-side, but that sounds like the sort of things you don't prioritize.)

Reply:
The per-attribute operations are available internally, and for the inline attribute, they are still exposed.  It's just that they were two call sites dealing with function attributes, and it thus is made sense to export one function to deal with all function attributes to avoid the duplicattion.  (Moreover, the new function deals with reporting the use of both `inline` and `local`.) Then I dropped function unused outside the module from its interface, but I can definitely restore that.

Reply:
I think it would be nicer if the per-attribute functions were available again. That would probably suffice as a change here.

Reply:
Done.

Reply:
Thanks! My reasoning on this part is that downstream users or non-expert contributors to the compiler codebase tend to hesitate before asking more functions to be exposed from an interface. Keeping things private is a good default, but here it is reasonable to assume that this logic will be needed by others, possibly in a compiler-libs context, so we shouldn't put barriers to doing the right thing.